src/main/java/org/altbeacon/beacon/service/scanner/CycledLeScanner.java

```diff
@@ -193,14 +223,14 @@ public void run() {
 
     protected abstract void startScan();
 
-    @SuppressLint("NewApi")
+    @MainThread
     protected void scanLeDevice(final Boolean enable) {
         try {
             mScanCyclerStarted = true;
             if (getBluetoothAdapter() == null) {
                 LogManager.e(TAG, "No Bluetooth adapter.  beaconService cannot scan.");
             }
-            if (enable) {
+            if (enable && mScanningEnabled) {
```

Comment:
Why would we want to allow a scan to begin after `stop` has already been called?

`mScanningEnabled` is the main "is this cycled scanner started or stopped" flag. It is only set in `start` and `stop`.